### ISSUE TITLE
Bulk update using array?

### ISSUE THREAD

**Issue by ryanwalters** [robconery/massive-js#252]
Hi there, new to massive and had a quick question.

Is it possible to bulk update? I know it is possible to bulk insert by passing in an array, but passing an array to `update` doesn't seem to work. If not, what is the recommended way of doing a bulk update?

**Comment by xivSolutions** [comment]
I believe you can call `update` directly (instead of using the `save` function, which relies on the presence of a primary key to determine whether to insert of update), and set the criteria as needed, and pass the values you wish to set the fields to. 

Let's think about the notion of a "bulk update" for a minute. To me, this would mean (in SQL terms) something like the following:

```SQL
UPDATE presidential_candidates SET is_viable  = FALSE WHERE is_valid = TRUE;
```

That SQL will set the `is_viable` field for all the records in the `presidential_candidates` table to `FALSE`

Likewise, from Massive, you would do this:

```js
db.presidential_candidates.update({is_valid: true}, {is_valid: false}, function(err, res) {
  // Do something after updating
});
```

What you cannot do with Massive (at least, at the moment) is pass an array of objecets, which *each need to be updated independently*. A couple of things to note here:

- The notion of "iterating over an array and updating things" is a very OO concept - it is not a SQL concept. Whether or not the `update method in Massive accepts an array, and then does some iteratin', or your client code does the iteratin' and then calls `update` for each, there is still some iteratin' goin' on. 

- Unlike the bulk insert method, in which Massive pretty much takes a batch of data from an array, and builds an SQL statement that doesn't require criteria, `update` needs to know which specific record(s) to update, and which fields to apply which values to. Building an SQL batch to do this is a whole different deal, and is not currently implemented in Massive. 

- Massive is not an ORM in the same sense as Entity Framework, NHibernate, or ActiveRecord - it doesn't do the change-tracking bits that make the sort of `update(someArrayOfObjects)` stuff possible. Massive does it's best to provide some essential query tools, and stay out of the way. Thus, it stays simple, but occasionally, you may find yourself thinking you should be able to do certain ORM-ey things with it that are not part of the design scope here. 

Long story short, if you're looking to do a "bublk update" in the SQL context ("update all the records that meet this criteria, and change these fields on each to these new values") then you're in luck - YOU CAN!!

If you are looking to do the OO version ("Here's a bunch of objects from this table where I changes some values on each one - store the changes") you can't, at least for now. At least, not in any way besides looping your objects and setting up a call to update for each one. 

Hope that helped. Let me know if youo have any questions, and also, skim through the test files to get a basic sense of how some of these calls work (if you haven't already) :-)

**Comment by ryanwalters** [comment]
That pretty much answered my question perfectly! I was looking for the "OO version", maybe it'll be supported one day.

Thanks.
